src/App.ts

```diff
@@ -260,34 +260,32 @@ export default class App {
       }
     }
 
-    let usingBuiltinOauth = false;
-    if (
-      clientId !== undefined
-      && clientSecret !== undefined
-      && (stateSecret !== undefined || (installerOptions !== undefined && installerOptions.stateStore !== undefined))
-      && this.receiver instanceof ExpressReceiver
-    ) {
-      usingBuiltinOauth = true;
+    let usingOauth = false;
+    if ((this.receiver as ExpressReceiver).installer !== undefined
```

Comment:
just trying to play out a scenario in my head: let's say someone passes in a custom receiver that happens to have some property called `installer`, which also has some property called `authorize` that has nothing to do with installation support in the way that we describe, what happens?

are we intentionally leaving the implementation of those properties open to developers that bring their own custom receiver? or, do we expect those developers to use our specific `InstallationProvider` class (which has no dependencies on any framework, since it only uses node's `req` and `res`)? if the latter is true, then could we settle for a more strict check, maybe `typeof this.receiver.installer === 'InstallProvider'`?

Reply:
Yeah I was worried that people could 'fake' their way through these checks. Tying it to actually use `InstallProvider` is a good way to ensure it is using the OAuth library and functions we expect under the hood.

I guess the question here is are we okay with folks implementing OAuth without our OAuth package? 

I'm going to merge this in for now but I think it is worth discussing.